testresultparser/parser_test.go

```diff
@@ -0,0 +1,154 @@
+				expectedResult := &TestResult{TestSuite: "com.udacity.gradle.test.PersonTest", Tests: 1, Failures: 1, Errors: 0, Skipped: 0, Time: "0.003", Failure: f}
+
+				// when
+				filepath := getFilepath("gradle_failure_test.xml")
+				myParser, _ := CreateParser(filepath, "surefire")
+				actualResult := myParser.Parse(filepath)
+
+				//then
+				Expect(actualResult).To(Equal(expectedResult))
+			})
+		})
+
+	})
+	Describe("Parsing TXT file", func() {
+		Context("With Maven-surefire plugin and TestNG", func() {
+			It("should return output struct", func() {
+				// given
+				f := []TestFailure{{TestCase: "foo", Type: "java.lang.AssertionError", Message: "expected [foo] but found [1.0]"}}
+				expectedResult := &TestResult{TestSuite: "testNgMavenExample1.TestNgMavenExampleTest", Tests: 2, Failures: 1, Errors: 0, Skipped: 0, Time: "0.785", Failure: f}
+
+				// when
+				filepath := getFilepath("testNG_failure_test.txt")
+				myParser, _ := CreateParser(filepath, "surefire")
+				actualResult := myParser.Parse(filepath)
+
+				//then
+				Expect(actualResult).To(Equal(expectedResult))
+			})
+		})
+		Context("With Maven-surefire plugin and TestNG and multiple failures", func() {
+			It("should return output struct", func() {
+				// given
+				f := []TestFailure{{
+					TestCase: "bar",
+					Type:     "java.lang.AssertionError",
+					Message:  "expected [foo] but found [2.0]",
+				},
+					{
+						TestCase: "foo",
+						Type:     "java.lang.AssertionError",
+						Message:  "expected [foo] but found [1.0]",
+					},
+				}
+				expectedResult := &TestResult{TestSuite: "testNgMavenExample1.TestNgMavenExampleTest", Tests: 4, Failures: 2, Errors: 0, Skipped: 1, Time: "1.395", Failure: f}
+
+				// when
+				filepath := getFilepath("testNG_failure_test2.txt")
+				myParser, _ := CreateParser(filepath, "surefire")
+				actualResult := myParser.Parse(filepath)
+
+				// then
+				Expect(actualResult).To(Equal(expectedResult))
+			})
+		})
+	})
+})
+
+// --- Test helpers
+
+func getFilepath(filename string) string {
+	_, packagefilename, _, _ := runtime.Caller(0)
```

Comment:
We should handle error here I guess (last argument) and panic if something went wrong.